### PR TITLE
test: Tests Jax  distributor components

### LINKED ISSUE #523: [TEST] Jax distributor
### What do you want to test?
Jax distributor components

### Outline of test structure
* Unit tests
* Test components and hooks

### Definition of done
Passing checks, cover all hooks, edge cases considered

### Mandatory checklist before making a PR
* [ ] The success criteria laid down in “Definition of done” are met.
* [ ] Test code is documented - docstrings for methods and classes, static types for arguments.
* [ ] Documentation is updated - README, CONTRIBUTING, or other documentation.

### PR DESCRIPTION
**Pull request by OmaymaMahjoub**
## What?
Unit tests for Jax  distributor components
## Why?
Components are currently not tested. Needs to be robust for Jax Mava release
## How?
Test `Distributor `component from `components/jax/building/distributor.py`
## Extra
Close #523

### CODE REVIEW

File: tests/jax/components/building/distributor_test.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    assert (
+        mock_builder.store.program._program._groups["executor"][-1]._constructor
+        == mock_builder.executor
+    )
+    assert (
+        mock_builder.store.program._program._groups["trainer"][-1]._constructor
+        == mock_builder.trainer
+    )
+
+    assert not "evaluator" in mock_builder.store.program._program._groups.keys()
```
Comment (RuanJohn) [inline tests/jax/components/building/distributor_test.py:196]: It is okay to remove this assertion since the assert on line 171 already verifies this.

Comment (OmaymaMahjoub) [reply tests/jax/components/building/distributor_test.py:196]: Okay, thanks for noticing it 🙌

**Review by RuanJohn — DISMISSED** [review]
All good from my side. Thank you @OmaymaMahjoub!

**Review by mmorris44 — APPROVED** [review]
Made a couple of naming changes, docstring changes, and made the distributor a fixture. Approved! Thanks @OmaymaMahjoub.
